src/test/java/uk/gov/pay/directdebit/mandate/resources/MandateResourceIT.java

```diff
@@ -87,70 +93,56 @@ public void tearDown() {
-        String returnUrl = "http://example.com/success-page/";
-        String postBody = new ObjectMapper().writeValueAsString(ImmutableMap.builder()
-                .put("return_url", returnUrl)
-                .build());
-
-        String requestPath = "/v1/api/accounts/{accountId}/mandates"
-                .replace("{accountId}", accountExternalId);
+        
+        Map<String, String> createMandateRequest = new HashMap<>();
+        Optional.ofNullable(returnUrl).ifPresent(x -> createMandateRequest.put("return_url", x));
+        Optional.ofNullable(serviceReference).ifPresent(x -> createMandateRequest.put("service_reference", x));
 
-        ValidatableResponse response = givenSetup()
-                .body(postBody)
-                .post(requestPath)
+        givenSetup()
+                .body(objectMapper.writeValueAsString(createMandateRequest))
+                .post("/v1/api/accounts/{accountId}/mandates".replace("{accountId}", accountExternalId))
                 .then()
-                .statusCode(Response.Status.CREATED.getStatusCode())
-                .body(JSON_MANDATE_ID_KEY, is(notNullValue()))
-                .body("return_url", is(returnUrl))
-                .body("created_date", is(notNullValue()))
-                .contentType(JSON);
-        MandateExternalId externalMandateId = MandateExternalId.valueOf(response.extract().path(JSON_MANDATE_ID_KEY).toString());
-
-        String documentLocation = expectedMandateLocationFor(accountExternalId, externalMandateId);
-        String token = testContext.getDatabaseTestHelper().getTokenByMandateExternalId(externalMandateId).get("secure_redirect_token").toString();
-
-        String hrefNextUrl = "http://Frontend/secure/" + token;
-        String hrefNextUrlPost = "http://Frontend/secure";
-
-        response
-                .body("links", hasSize(3))
-                .body("links", containsLink("self", "GET", documentLocation))
-                .body("links", containsLink("next_url", "GET", hrefNextUrl))
-                .body("links", containsLink("next_url_post", "POST", hrefNextUrlPost, "application/x-www-form-urlencoded", new HashMap<String, Object>() {{
-                    put("chargeTokenId", token);
-                }}));
-
-        Map<String, Object> createdMandate = testContext.getDatabaseTestHelper().getMandateByExternalId(externalMandateId);
-
-        assertThat(createdMandate.get("external_id"), is(notNullValue()));
-        assertThat(createdMandate.get("return_url"), is(returnUrl));
-        assertThat(createdMandate.get("gateway_account_id"), is(gatewayAccountFixture.getId()));
+                .statusCode(HttpStatus.SC_UNPROCESSABLE_ENTITY)
+                .body("message", contains(expectedErrorMessage))
+                .body("error_identifier", is("GENERIC"));
     }
 
     @Test
-    public void shouldCreateAMandate_withAllFields() throws Exception {
+    @Parameters({"null", "raindrops on roses and whiskers on kittens"})
```

Comment:
🎶 Bright copper kettles and warm woolen mittens 🎶  (I had to google that...)

Reply:
my favorite tings

Reply:
https://www.youtube.com/watch?v=3pzIlq7jZzw